pages/en/contrib/style-guide.md

```diff
@@ -449,6 +449,44 @@ async function getDodo() {
 }
 ```
 
+### Always await the last promise in a function
+
+If the last line of a function handles a promise, always await that promise, regardless whether the result will be returned or not. (This makes it clear whether we did or did not intend to return the value, and dismisses any idea that we are still linking up the promise chain in pre-await style.)
+
+**Good**:
+
+```js
+async function openCircus() {
+  await openCarPark();
+  await openDoors();
+}
+```
+
+**Good**:
+
+```js
+async function getAnimal() {
+  return await getElephant();
+}
+```
+
+**Bad**:
+
+```js
+async function openCircus() {
+  await openCarPark();
+  return openDoors();
+}
+```
+
+**Bad**:
+
+```js
+async function getAnimal() {
+  return getElephant();
```

Comment:
In this case, `async` keyword is not needed as no `await` is used in the method body.

Reply:
I think the point @joeytwiddle was trying to give a bad example of using promises and async/await. Users might be tempted to return a promise for the purpose of promise chaining, so this example is showing them to not do that

Reply:
I understand. But I would like to help users understand why it's bad.

Reply:
In this example I want to emphasize that the mistake is the lack of `await`. If we have an example without `async` I am worried the reader may think that the missing `async` is what is wrong.

It's worth noting that two sections above we said "Always use the `async` keyword" so not having `async` is already considered a mistake.

Reply:
That's fine.